sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java

```diff
@@ -115,7 +116,7 @@ public final class AutoConfiguredOpenTelemetrySdkBuilder implements AutoConfigur
    * {@link #addPropertiesSupplier(Supplier)} and {@link #addPropertiesCustomizer(Function)} will
    * have no effect if this method is used.
    */
-  AutoConfiguredOpenTelemetrySdkBuilder setConfig(ConfigProperties config) {
+  public AutoConfiguredOpenTelemetrySdkBuilder setConfig(ConfigProperties config) {
```

Comment:
Can you explain the reasoning for this? Its pretty drastic to want to call this and negate calls to `addPropertiesSupplier` and `addPropertiesCustomizer`.

Its currently only being used in tests, and even those usages aren't necessary. I honestly think we ought to delete it altogether rather than promote it to the public API.

Reply:
My suggestion: Make this method experimental like `setComponentLoader`

Reply:
> My suggestion: Make this method experimental like setComponentLoader

What would be the path to stabilizing? I think we should stabilize `setComponentLoader` after we integrate it into spring autoconfigure and confirm there we didn't miss anything. But overriding the config here conflicts with autoconfigure patterns by erasing property suppliers and customizers provided via SPI. Overriding ConfigProperties means that spring autoconfigure would interact with _most_ SPIs normally, except those involving customizing config properties. Would be surprising to users.

Reply:
What about an experimental method to customize ConfigProperties? Something like:
```
AutoConfiguredOpenTelemetrySdkBuilder setConfigPropertiesCustomizer(Function<ConfigProperties, ConfigProperties> configPropertiesCustomizer)
```
The idea is that it follows a similar pattern to the customization methods in `AutoConfigurationCustomizer` like `AutoConfigurationCustomizer#addPropertiesCustomizer(Function<ConfigProperties, Map<String, String>>)`, but is able to return a custom `ConfigProperties` instance instead of `Map<String, String>`. We can limit this method to `AutoConfigurationOpenTelemetrySdkBuilder` (i.e. do not add it to `AutoConfigurationCustomizer`). Spring autoconfiguration's implementation of `ConfigProperties` can fallback to the `ConfigProperties` instance provided by autoconfigure if a particular property is undefined in spring's configuration system.

Reply:
That idea is quite elegant.

I don't think a spring user would expect that capability, because spring can already do that: https://stackoverflow.com/questions/29072628/how-can-i-override-spring-boot-application-properties-programmatically

Reply:
I think your proposal could be good for distros of the spring starter, so I'll implement it.